### ISSUE TITLE
QScript convert a `ShiftArray` query reading from the file system

### ISSUE THREAD

**Issue by alissapajer** [quasar-analytics/quasar#1544]
```scala   
 "convert a read shift array" in {
      // select (baz || quux || ducks)[*] from `/foo/bar`
      convert(
        None,
        LP.Let('x, lpRead("/foo/bar"),
          structural.ShiftArray(
            structural.ArrayConcat(
              structural.ArrayConcat(
                structural.ObjectProject(LP.Free('x), LP.Constant(Data.Str("baz"))).embed,
                structural.ObjectProject(LP.Free('x), LP.Constant(Data.Str("quux"))).embed).embed,
              structural.ObjectProject(LP.Free('x), LP.Constant(Data.Str("ducks"))).embed).embed).embed)) must
      equal(chain(RootR).some) // TODO incorrect expectation
    }
```
The previous query should compile to correct qscript.

**Comment by alissapajer** [comment]
This test is `pendingUntilFixed`.

**Comment by sellout** [comment]
Fixed in #1574.
